metricbeat/module/elasticsearch/index_summary/data.go

```diff
@@ -66,25 +66,24 @@ var (
 	}
 )
 
-func eventMapping(r mb.ReporterV2, info elasticsearch.Info, content []byte) []error {
+func eventMapping(r mb.ReporterV2, info elasticsearch.Info, content []byte) error {
 	var all struct {
 		Data map[string]interface{} `json:"_all"`
 	}
 
 	err := json.Unmarshal(content, &all)
 	if err != nil {
 		r.Error(err)
```

Comment:
One nice thing that we can do now is that we can check for the error from `eventMapping` and report it in only 1 place. The current code with reporting and returning the error now is only returning the error as there is only 1 type of errors left.